src/main/java/org/spongepowered/api/service/ServiceManager.java

```diff
@@ -48,14 +50,14 @@
      * <p>Services should only be registered during initialization. If services
      * are registered later, then they may not be utilized.</p>
      *
-     * @param plugin The instance of a plugin
+     * @param cause The cause of the provider change.
      * @param service The service
      * @param provider The implementation
      * @param <T> The type of service
      * @throws IllegalArgumentException Thrown if {@code plugin} is not a
      *     plugin instance
      */
-    <T> void setProvider(Object plugin, Class<T> service, T provider);
+    <T> void setProvider(Cause cause, Class<T> service, T provider);
```

Comment:
What else do you want to specify using Cause?
IMO a plugin just provides a Service. There is no cause like `Player x entered y`.
And well the cause `plugin is installed` does not count.